backbone.js

```diff
@@ -1140,6 +1140,7 @@
         removed.push(model);
         this._removeReference(model, options);
       }
+      if (models.length > 0 && !options.silent) delete options.index;
```

Comment:
@akre54 - Coincidentally, came across a use case on a chained action.  A copy of the `options` with defaults for unset properties was created in the event listener.  The `options` copy was used to modify a related collection with different `set` behavior.

`options = _.extend({index:0}, options);`


The default, won't take when `options.index = void 0`;